pkg/ccl/backupccl/restore_job.go

```diff
@@ -1064,6 +1071,49 @@ func createImportingDescriptors(
 				}
 			}
 
+			// In a cluster restore, we do not restore the database `defaultdb` from
+			// the BACKUP, but instead re-parent all backed up schema objects that
+			// were in `defaultdb` to point to the restoring cluster's `defaultdb`.
+			// This special handling is to account for the fact that every new cluster
+			// has a `defaultdb` at a static ID created on startup.
+			// As a result of this, we must set the RegionConfig of `defaultdb` on
+			// the restoring cluster, to match the config set on the backed up
+			// `defaultdb` descriptor.
+			if details.DescriptorCoverage == tree.AllDescriptors {
+				if backedUpDefaultDBDesc != nil && backedUpDefaultDBDesc.IsMultiRegion() {
+					// Lookup the `defaultdb` in the restoring cluster.
+					defaultDBID, err := lookupDatabaseID(ctx, txn, p.ExecCfg().Codec,
+						catalogkeys.DefaultDatabaseName)
+					if err != nil {
+						return err
+					}
+					backedUpDefaultDBRegionConfig := backedUpDefaultDBDesc.GetRegionConfig()
+					for _, typ := range typesByID {
+						// Find the multi-region enum that is referenced by the backed up
+						// `defaultdb` and remap it to point to the restored multi-region
+						// enum. Since this is a cluster restore, we can expect the
+						// `defaultdb` zone config to be restored as part of the
+						// system.zones table.
+						if typ.GetParentID() == defaultDBID && typ.GetKind() == descpb.TypeDescriptor_MULTIREGION_ENUM {
+							backedUpDefaultDBRegionConfig.RegionEnumID = typ.GetID()
+							break
+						}
```

Comment:
fortunately, we validate that the region enum ID we are pointing to actually refers to a type in the database - https://github.com/adityamaru/cockroach/blob/4a3ad27613d88c02ddfe8d4726c25b5d8411186f/pkg/sql/catalog/dbdesc/database_desc.go#L260-L260. So if this is set to something non-sensical it will error out when we go to write the database descriptor below.

Your comment got me thinking though that we don't really need this mapping since in a cluster restore, a type will have the same ID as when it was backed up. Removed the for loop entirely.